source/packages/libraries/core/deployment-helper/src/customResources/eventSource.customResource.ts

```diff
@@ -0,0 +1,108 @@
+/*-------------------------------------------------------------------------------
+# Copyright (c) 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# This source code is subject to the terms found in the AWS Enterprise Customer Agreement.
+#-------------------------------------------------------------------------------*/
+import { inject, injectable } from 'inversify';
+
+import { logger } from '../utils/logger';
+
+import {CustomResourceEvent} from './customResource.model';
+import { LambdaInvokerService, LAMBDAINVOKE_TYPES, LambdaApiGatewayEventBuilder } from '@cdf/lambda-invoke';
+import { CustomResource } from './customResource';
+import ow from 'ow';
+
+@injectable()
+export class EventSourceCustomResource implements CustomResource {
```

Comment:
I did some research and it would be difficult to split eventSource creation and event creation as the events depend on the existence of the eventSource. If the eventSource is created without ID specified a UUID will be generated in which case the associate events will be difficult/impossible to create.

Reply:
> The Name implies there is a single eventsource which gets created but this custom resources adds them in bulk. This should be renamed to `EventSources`.

This resource creates 1 eventSource with multiple events associated with it. Should I change its class to EventSourcesCustomResource?

Reply:
Can we approve it since it aligns with the current customResouces?

Reply:
Commands custom resource is an internal helper which configures a Iot job and other components which needs to be deployed as part of the service. Its abstracted away and is not utilized or should'nt be used externally. 

External resources needs to be designed for scalability. In this case, splitting the eventsources and events into 2 resources provides more flexibility. May be a user wants to just define the event sources but not events.

The EventSource custom resource can easily output the eventsource Id which can be reference within cloudformation as an output of the custom resource and then specified in the EventSourceEvents Custom resource.